File: backend/Models/Bills.cs
namespace MyApi.Models;

public class Bills
{
    public int Id { get; set; }

    public required int TableId { get; set; }
    public Tables? Table { get; set; }

    // Khách trung thành gắn lúc thanh toán (nullable — guest không có)
    public int? UserId { get; set; }
    public User? User { get; set; }

    // Open / Paid / Cancelled
    public required string Status { get; set; } = "Open";

    public decimal Subtotal { get; set; }         // cộng từ tất cả Orders trong bill
    public decimal DiscountAmount { get; set; }   // giảm 10% khi dùng điểm
    public int LoyaltyPointsUsed { get; set; }    // 400 điểm đã tiêu
    public decimal Total { get; set; }            // Subtotal - DiscountAmount

    public int LoyaltyPointsEarned { get; set; }  // +100 điểm sau khi thanh toán

    public DateTime CreatedAt { get; set; }
    public DateTime? PaidAt { get; set; }

    public List<Orders> Orders { get; set; } = new();
}

File: frontend/src/api/bills.ts
import { api } from "./client";
import type { OrderResponse } from "./orders";

export interface Bill {
  id: number;
  tableId: number;
  userId: number | null;
  status: "Open" | "Paid" | "Cancelled";
  subtotal: number;
  discountAmount: number;
  loyaltyPointsUsed: number;
  total: number;
  loyaltyPointsEarned: number;
  createdAt: string;
  paidAt: string | null;
  orders: OrderResponse[];
}

export interface BillListResponse {
  items: Bill[];
  total: number;
  page: number;
  limit: number;
}

/** Danh sách hoá đơn (dashboard) — yêu cầu đăng nhập. */
export async function getBills(page = 1, limit = 10): Promise<BillListResponse> {
  const { data } = await api.get<BillListResponse>("/bills", {
    params: { page, limit },
  });
  return data;
}

export async function getBill(id: number): Promise<Bill> {
  const { data } = await api.get<Bill>(`/bills/${id}`);
  return data;
}

export interface PayBillResult {
  message: string;
  bill: Bill;
  pointsEarned: number;
}

/**
 * Thanh toán hoá đơn.
 * Khi `useLoyaltyPoints = true`, khách (có userId) dùng 400 điểm để giảm 10%.
 */
export async function payBill(
  id: number,
  useLoyaltyPoints = false,
): Promise<PayBillResult> {
  const { data } = await api.post<PayBillResult>(`/bills/${id}/pay`, {
    useLoyaltyPoints,
  });
  return data;
}